src/MutationObserver.js

```diff
@@ -138,27 +131,44 @@ class MutationObserver {
   }
 
   handleAttribute(el, name, value, oldValue) {
-    this.queue.push(new MutationRecord('attributes', el, null, null, null, null, name, null, oldValue));
+    // Respect attribute filter.
+    if (this.options.attributeFilter && !this.options.attributeFilter.includes(name)) {
+      return;
+    }
 
-    setImmediate(() => {
-      this.flush();
-    });
+    this.queue.push(new MutationRecord('attributes', el, emptyNodeList, emptyNodeList,
+                                       null, null, name, null, oldValue));
+    setImmediate(() => { this.flush(); });
   }
 
   handleChildren(el, addedNodes, removedNodes, previousSibling, nextSibling) {
-    this.queue.push(new MutationRecord('childList', el, addedNodes, removedNodes, previousSibling, nextSibling, null, null, null));
+    // Include recursive children in added and removed nodes list.
+    if (addedNodes) {
+      for (let i = 0; i < addedNodes.length; i++) {
+        addedNodes[i].traverse(node => {
+          if (node.nodeType !== Node.ELEMENT_NODE || addedNodes.includes(node)) { return; }
```

Comment:
I think if we traverse starting at the `childNodes` then we don't need this `.includes`. I think an _actual_ DOM loop would be a serious bug.